crates/egui/src/widgets/text_edit/builder.rs

```diff
@@ -922,7 +944,16 @@ fn events(
                 // Newlines are handled by `Key::Enter`.
                 if !text_to_insert.is_empty() && text_to_insert != "\n" && text_to_insert != "\r" {
                     let mut ccursor = delete_selected(text, &cursor_range);
-                    insert_text(&mut ccursor, text, text_to_insert);
+
+                    match char_limit {
+                        Some(limit) if !multiline => {
+                            let mut new_string = text_to_insert.clone();
+                            new_string.truncate(limit - text.as_str().len());
```

Comment:
This chops UTF-8 sequences at the middle, causing panics.

```
thread 'main' panicked at 'assertion failed: self.is_char_boundary(new_len)', /rustc/897e37553bba8b42751c67658967889d11ecd120/library/alloc/src/string.rs:1280:13
stack backtrace:
   ...
   4: egui::widgets::text_edit::builder::events
             at ./crates/egui/src/widgets/text_edit/builder.rs:951:29
   ...
```

Reply:
Thank you for bringing this up, I have just fixed it.
😃